design/cdk-bootstrap.md

```diff
@@ -0,0 +1,444 @@
+  if they encounter any errors.
+
+## Bootstrap template
+
+Here is the JSON of the bootstrap CloudFormation template:
+
+```json
+{
+  "Parameters": {
+    "ResourceNamePrefix": {
+      "Description": "The prefix to use for all of the physical names of the bootstrap resources",
+      "Default": "aws-cdk-bootstrap-hnb659fds",
+      "Type": "String"
+    },
+    "StagingBucketName": {
+      "Description": "The name of the S3 bucket used to store the file assets (without the prefix and suffix)",
+      "Default": "staging-bucket",
+      "Type": "String"
+    },
+    "StagingBucketKmsKeyId": {
+      "Description": "The identifier of the KMS Key used to encrypt the contents of the ",
+      "Default": "",
+      "Type": "String"
+    },
+    "DockerAssetsStagingRepositoryName": {
+      "Description": "The name of the ECR Docker repository used to store the Docker assets (without the prefix and suffix)",
+      "Default": "docker-assets-repository",
+      "Type": "String"
+    },
+    "TrustedAccounts": {
+      "Description": "List of AWS account IDs that the publish and action roles should trust to be assumed from",
+      "Default": "",
+      "Type": "CommaDelimitedList"
+    },
+    "CfnDeploymentRoleManagedPolicies": {
+      "Description": "List of the ManagedPolicy ARN(s) to attach to the CloudFormation deployment role",
+      "Default": "",
+      "Type": "CommaDelimitedList"
+    }
+  },
+  "Conditions": {
+    "HasStagingBucketKmsKeyId": {
+      "Fn::Not": [
+        {
+          "Fn::Equals": [
+            "",
+            { "Ref": "StagingBucketKmsKeyId" }
+          ]
+        }
+      ]
+    },
+    "HasTrustedAccounts": {
+      "Fn::Not": [
+        {
+          "Fn::Equals": [
+            "",
+            {
+              "Fn::Join": [
+                "",
+                {
+                  "Ref": "TrustedAccounts"
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  },
+  "Resources": {
+    "StagingBucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": {
+          "Fn::Sub": "${ResourceNamePrefix}-${StagingBucketName}-${AWS::AccountId}-${AWS::Region}"
+        },
+        "AccessControl": "Private",
+        "BucketEncryption": {
+          "ServerSideEncryptionConfiguration": [{
+            "ServerSideEncryptionByDefault": {
+              "SSEAlgorithm": "aws:kms",
+              "KMSMasterKeyID": {
+                "Fn::If": [
+                  "HasStagingBucketKmsKeyId",
+                  { "Ref":  "StagingBucketKmsKeyId" },
+                  { "Ref":  "AWS::NoValue" }
+                ]
+              }
+            }
+          }]
+        },
+        "PublicAccessBlockConfiguration": {
+          "BlockPublicAcls": true,
+          "BlockPublicPolicy": true,
+          "IgnorePublicAcls": true,
+          "RestrictPublicBuckets": true
+        }
+      }
+    },
```

Comment:
ContainerAssetsRepository (not need “staging” which has another meaning in the context of the cloud assembly)